src/tests/src/solver/optim/api/testApiOrtoolsLinearProblem.cpp

```diff
@@ -133,108 +151,87 @@ BOOST_FIXTURE_TEST_CASE(mipVariableBounds, Fixture)
 
     BOOST_CHECK_EQUAL(var->getLb(), 2);
     BOOST_CHECK_EQUAL(var->getUb(), 13);
-
-    auto* sameVar = pb->getVariable("var");
-    BOOST_CHECK_EQUAL(sameVar->getLb(), 2);
-    BOOST_CHECK_EQUAL(sameVar->getUb(), 13);
 }
 
-BOOST_FIXTURE_TEST_CASE(mipConstraintBounds, Fixture)
+BOOST_FIXTURE_TEST_CASE(give_bounds_to_constraint___check_bounds_exist, FixtureEmptyProblem)
 {
     auto* constraint = pb->addConstraint(0, 1, "var");
 
     constraint->setLb(-4);
     constraint->setUb(7);
-
     BOOST_CHECK_EQUAL(constraint->getLb(), -4);
     BOOST_CHECK_EQUAL(constraint->getUb(), 7);
 
     constraint->setBounds(2, 13);
-
     BOOST_CHECK_EQUAL(constraint->getLb(), 2);
     BOOST_CHECK_EQUAL(constraint->getUb(), 13);
 }
 
-BOOST_FIXTURE_TEST_CASE(objectiveCoeff, Fixture)
+BOOST_FIXTURE_TEST_CASE(give_cost_to_variable___check_cost_exist, FixtureEmptyProblem)
 {
     auto* var = pb->addVariable(0, 1, true, "var");
     pb->setObjectiveCoefficient(var, 1);
     BOOST_CHECK_EQUAL(pb->getObjectiveCoefficient(var), 1);
+}
 
+BOOST_FIXTURE_TEST_CASE(get_cost_from_null_variable_leads_to_bad_cast, FixtureEmptyProblem)
+{
     BOOST_CHECK_THROW(pb->getObjectiveCoefficient(nullptr), std::bad_cast);
-    BOOST_CHECK_THROW(pb->setObjectiveCoefficient(nullptr, 0), std::bad_cast);
 }
 
-BOOST_FIXTURE_TEST_CASE(infeasibleProblem, Fixture)
+BOOST_FIXTURE_TEST_CASE(give_cost_to_null_variable_leads_to_bad_cast, FixtureEmptyProblem)
 {
-    auto* var1 = pb->addIntVariable(0, 10, "var1");
-    auto* var2 = pb->addNumVariable(0, 10, "var2");
-    auto* constraint = pb->addConstraint(1, 1, "constraint");
-
-    constraint->setCoefficient(var1, 1);
-    constraint->setCoefficient(var2, 1);
-
-    pb->setObjectiveCoefficient(var1, 1);
-    pb->setObjectiveCoefficient(var2, 1);
-    pb->setMinimization();
+    BOOST_CHECK_THROW(pb->setObjectiveCoefficient(nullptr, 0), std::bad_cast);
+}
 
+BOOST_FIXTURE_TEST_CASE(solve_infeasible_problem_leads_to_error_status, FixtureInfeasibleProblem)
+{
     auto* solution = pb->solve(true);
-
     BOOST_CHECK(solution->getStatus() == Api::MipStatus::MIP_ERROR);
```

Comment:
looking back at this, I think we should add a MipStatus::INFEASIBLE value, and not treat it as a default MIP_ERROR

Reply:
Added **MipStatus::INFEASIBLE** item, as asked